src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs

```diff
@@ -149,54 +149,122 @@ protected override void HandleTaskExecutionCore()
+        private void GetRenderPackagesFromMirrorDataImp(
+            Guid outputPortId,
+            MirrorData mirrorData,
+            IRenderPackage package,
+            string labelKey)
+        {
+
             if (mirrorData.IsCollection)
             {
                 int count = 0;
                 foreach (var el in mirrorData.GetElements())
                 {
                     if (el.IsCollection || el.Data is IGraphicItem)
                     {
-                        string newTag = tag + ":" + count;
-                        GetRenderPackagesFromMirrorData(outputPortId, el, newTag, displayLabels);
+                        string newLabel = labelKey + ":" + count;
+                        GetRenderPackagesFromMirrorDataImp(outputPortId, el, package, newLabel);
                     }
-                    count = count + 1;
+                    count += 1;
                 }
             }
             else
             {
-                var graphicItem = mirrorData.Data as IGraphicItem;
-                if (graphicItem == null)
+                if (!(mirrorData.Data is IGraphicItem graphicItem))
                 {
                     return;
                 }
 
-                var package = factory.CreateRenderPackage();
-                var packageWithTransform = package as ITransformable;
-                package.Description = tag;
+                GetRenderPackageFromGraphicItem(outputPortId, graphicItem, package, labelKey);
+            }
+        }
+
+        private void GetRenderPackageFromGraphicItem(Guid outputPortId, IGraphicItem graphicItem, IRenderPackage package, string labelKey)
+        {
+            var packageWithTransform = package as ITransformable;
+
+            try
+            {
+                var previousPointVertexCount = package.PointVertexCount;
+                var previousLineVertexCount = package.LineVertexCount;
+                var previousMeshVertexCount = package.MeshVertexCount;
 
-                try
+                //Plane tessellation needs to be handled here vs in LibG currently
+                if (graphicItem is Plane plane)
+                {
+                    CreatePlaneTessellation(package, plane);
+                }
+                else
                 {
-                    graphicItem.Tessellate(package, factory.TessellationParameters);
-                    if (package.MeshVertexColors.Any())
+                    try
                     {
-                        package.RequiresPerVertexColoration = true;
+                        graphicItem.Tessellate(package, factory.TessellationParameters);
                     }
-
-                    //If the package has a transform that is not the identity matrix
-                    //then set requiresCustomTransform to true.
-                    if (packageWithTransform != null && packageWithTransform.Transform.SequenceEqual(
-                        new double[] { 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 }) == false)
```

Comment:
I think that case is valid.... I just don't think that is what our implementation supports currently.  Today you add your tessellated geometry and then you can additionally add a transform.  That transform does not support instancing it only transforms all the geometry in the render package.  I think the original use case being able to tesselate an object once and then move it around.

Reply:
I think instancing would be much more interesting and we should add that as an enhancement.  I think we should do it as a follow up PR.  It would need a similar data structure to the color aspects where you can apply numerous transforms to any range of geometry data.  The advantage would be that we could and tessellated geometry data once and supply a transform matrix for each instance.  There would also need to changes to the helix implementation to support real instancing